src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/FindReferencesWindowInProcess.cs

```diff
@@ -61,9 +63,9 @@ private async Task<IWpfTableControl2> GetFindReferencesWindowAsync(CancellationT
         {
             await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
-            var dte = await GetRequiredGlobalServiceAsync<SDTE, EnvDTE80.DTE2>(cancellationToken);
-
-            var toolWindow = dte.ToolWindows.GetToolWindow(FindReferencesWindowGuid);
+            var shell = await GetRequiredGlobalServiceAsync<SVsUIShell, IVsUIShell>(cancellationToken);
```

Comment:
just for my own curiosity, why is this way preferred over dte?  I don't know too much about either

Reply:
> why is this way preferred over dte

DTE is a slow wrapper around other public APIs. It makes it hard to tell what's happening under the covers, so removing this indirection step helps with both performance and clarity.